packages/rocket-batch-file-process-aws-infrastructure/src/batch-file-processing.ts

```diff
@@ -29,47 +36,71 @@ export class BatchFileProcessingStack {
 
     const eventsStore = stack.node.tryFindChild('events-store') as Table
 
-    const fileTriggerFunction = new Function(stack, 'rocketS3Trigger', {
+    const fileSplitterFunction = new Function(stack, 'fileSplitterFunction', {
       runtime: Runtime.NODEJS_12_X,
       timeout: Duration.minutes(15),
       memorySize: 1024,
       handler: 'index.handler',
-      functionName: config.appName + '-s3-rocket-trigger',
-      code: Code.fromAsset(path.join(__dirname, 'lambdas')),
+      functionName: config.appName + '-file-splitter-function',
+      code: Code.fromAsset(path.join(__dirname, 'file-splitter-lambda')),
       environment: {
         EVENT_STORE_NAME: eventsStore.tableName,
-        CHUNK_SIZE: params.chunkSize,
+        CHUNK_SIZE: params.config.chunkSize,
         STAGING_BUCKET_NAME: stagingBucketName,
         ENTITY_TYPE_NAME: 'File',
         TYPE_NAME: 'FileAdded',
       },
     })
 
-    fileTriggerFunction.addToRolePolicy(createPolicyStatement([eventsStore.tableArn], ['dynamodb:Put*']))
+    fileSplitterFunction.addToRolePolicy(createPolicyStatement([eventsStore.tableArn], ['dynamodb:Put*']))
 
-    sourceBucket.grantRead(fileTriggerFunction)
+    sourceBucket.grantRead(fileSplitterFunction)
 
-    fileTriggerFunction.addToRolePolicy(
+    fileSplitterFunction.addToRolePolicy(
       createPolicyStatement(
         [stagingBucket.bucketArn, stagingBucket.bucketArn + '/*'],
         ['s3:ListObject', 's3:PutObject', 's3:GetObject']
       )
     )
 
-    const uploadEvent = new S3EventSource(sourceBucket, {
+    const sourceUploadEvent = new S3EventSource(sourceBucket, {
       events: [EventType.OBJECT_CREATED],
     })
-    fileTriggerFunction.addEventSource(uploadEvent)
+    fileSplitterFunction.addEventSource(sourceUploadEvent)
 
     const eventsHandlerLambda = stack.node.tryFindChild('events-main') as Function
     eventsHandlerLambda.addToRolePolicy(
       createPolicyStatement([stagingBucket.bucketArn, stagingBucket.bucketArn + '/*'], ['s3:*'])
     )
+
+    const fileToLineEventFunction = new Function(stack, 'fileToLineEventFunction', {
```

Comment:
I have a question regarding these two functions: now we send 2 events right? (one per lambda)
1. One with `fileUri` and `fileSize`, that will trigger the `fileToLineEventFunction` lambda.
2. Another one with the jsonl or csv contents (`fileToLineEventFunction`).

If this is the case, shouldn't be this lambda (2) optional? Maybe I don't need this second one because I'm updating PDFs/images/txts to my S3 and I don't want a second lambda to be deployed nor triggered.

Reply:
The second lambda could be optional if you want to handle with Files from your booster app. The goal is this further abstraction is exactly to deal with that.
Previous version: Dealing with events representing the chunk files. You still need the SDK to open the files, split the data etc..
Curren version: you receive an event representing one line of the file, you even forget you are dealing with files.